pkg/discovery/dtofactory/general_builder.go

```diff
@@ -209,19 +217,30 @@ func (builder generalBuilder) metricValue(entityType metrics.DiscoveredEntityTyp
 	}
 
 	value := metric.GetValue()
-	switch value.(type) {
+	switch typedValue := value.(type) {
 	case []metrics.Point:
 		var sum float64
 		var peak float64
-		for _, point := range value.([]metrics.Point) {
+		for _, point := range typedValue {
 			sum += point.Value
 			peak = math.Max(peak, point.Value)
 		}
-		metricValue.Avg = sum / float64(len(value.([]metrics.Point)))
+		metricValue.Avg = sum / float64(len(typedValue))
+		metricValue.Peak = peak
+	case []metrics.ThrottlingCumulative:
+		throttled, total, peak, ok := aggregateContainerThrottlingSamples(entityID, typedValue)
+		if !ok {
+			// We don't have enough samples to calculate this value.
+			break
+		}
+
+		if total > 0 {
+			metricValue.Avg = throttled * 100 / total
```

Comment:
I know this is very unlikely, but what if throttled > total? Do we want to log a warning message and then set throttled = total in this case?

Reply:
Thanks for the suggestion @ading1977. Yes it makes sense to add this check. I am working on implementing the code to handle counter resets. This piece of code might change there. I will include this check also as part of the same.